src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+          },
+        ],
+      },
+      {
+        key: "line",
+        label: "Custom Line",
+        descriptionText:
+          "X Axis can only have attributes, Y Axis can only have measures, Color can only have attributes. " +
+          "Should have just 1 column in Y axis with colors columns.",
+        columnSections: [
+          {
+            key: "y",
+            label: "Custom Y Axis",
+            allowAttributeColumns: false,
+            allowMeasureColumns: true,
+            allowTimeSeriesColumns: false,
+          },
+          {
+            key: "legend",
+            label: "Slice with color",
+            allowAttributeColumns: true,
+            allowMeasureColumns: false,
+            allowTimeSeriesColumns: false,
+          },
+        ],
+      },
+      {
+        key: "stacked-column",
+        label: "Custom Stacked Column",
+        descriptionText:
+          "X Axis can only have attributes, Y Axis can only have measures, Color can only have attributes. " +
+          "Should have just 1 column in Y axis with colors columns.",
+        columnSections: [
+          {
+            key: "y",
+            label: "Custom Y Axis",
+            allowAttributeColumns: false,
+            allowMeasureColumns: true,
+            allowTimeSeriesColumns: false,
+          },
+          {
+            key: "legend",
+            label: "Slice with color",
+            allowAttributeColumns: true,
+            allowMeasureColumns: false,
+            allowTimeSeriesColumns: false,
+          },
+        ],
+      },
+      {
+        key: "scatter",
+        label: "Custom Scatter",
+        descriptionText:
+          "X Axis can only have attributes, Y Axis can only have measures, Color can only have attributes. " +
+          "Should have just 1 column in Y axis with colors columns.",
+        columnSections: [
+          {
+            key: "y",
+            label: "Custom Y Axis",
+            allowAttributeColumns: false,
+            allowMeasureColumns: true,
+            allowTimeSeriesColumns: false,
+          },
+          {
+            key: "legend",
+            label: "Slice with color",
+            allowAttributeColumns: true,
+            allowMeasureColumns: false,
+            allowTimeSeriesColumns: false,
+          },
+        ],
+      },
+    ],
+    visualPropEditorDefinition: {
+      elements: [
+        {
+          key: "datalabels",
+          type: "toggle",
+          defaultValue: false,
+          label: "Data Labels",
+        },
+        {
+          key: "hideLegend",
+          type: "toggle",
+          defaultValue: true,
+          label: "Legends",
+        },
+      ],
+    },
+    validateConfig: (updatedConfig, chartModel): ValidationResponse => {
+      if (updatedConfig.length <= 0) {
+        return {
+          isValid: false,
+          validationErrorMessage: ["invalid config. no config found"],
+        };
+      }
+      // assuming 0 is x dimension
```

Comment:
lets not assume and get dimension by key.

Reply:
done